versionedRouter.js

```diff
@@ -905,7 +907,9 @@ async function handleProxyRequest(destination, ctx) {
     });
     response = destNetworkHandler.responseHandler(
       processedProxyResponse,
-      destination
+      destination,
+      metadata.destInfo,
```

Comment:
Even if we were to take this approach, destructuring the metadata should happen within the response handler.